pkg/compliance/io.go

```diff
@@ -0,0 +1,249 @@
+package compliance
+
+import (
+	"context"
+	"fmt"
+	"github.com/aquasecurity/starboard/pkg/apis/aquasecurity/v1alpha1"
+	"github.com/aquasecurity/starboard/pkg/ext"
+	"github.com/emirpasic/gods/sets/hashset"
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+)
+
+type Mgr interface {
+	GenerateComplianceReport(ctx context.Context, spec v1alpha1.ReportSpec) (*v1alpha1.ClusterComplianceReport, error)
+}
+
+func NewMgr(client client.Client, log logr.Logger) Mgr {
+	return &cm{
+		client: client,
+		log:    log,
+	}
+}
+
+type cm struct {
+	client client.Client
+	log    logr.Logger
+}
+
+type summaryTotal struct {
```

Comment:
its used for populating data to compliance and complianceDetails report prefer not to couple it to compliance summary also its a mid logic calc I think it should populate data to summary at the end before publishing

Reply:
Interesting, but both structs have the same fields so I don't see any point in defining. Would you also define `internal_string` instead of using built-in `string` in Go?

Reply:
true it has the same fields now but we want to keep it flexible, if I want to change it , since it is used for mid processing I do not need to bread the CRD structure

Reply:
What flexibility we need to count totals? If v1alpha1.ComplianceSummary won't be enough then we can add custom struct.